src/core/ContactUser.h

```diff
@@ -135,11 +170,18 @@ private slots:
     void onSettingsModified(const QString &key, const QJsonValue &value);
 
 private:
+#ifdef PROTOCOL_NEW
+    // XXX be paranoid about tracking deletion of m_connection
```

Comment:
A commit in a later series addresses this by making m_connection a QPointer, and cleaning up its deletion logic a bit.